k8s/backend.yaml

```diff
@@ -0,0 +1,601 @@
+# send everything to http://otel-collector.observability-backend.svc.cluster.local:4317
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: observability-backend
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: collector-config
+  namespace: observability-backend
+data:
+  collector.yaml: |
```

Comment:
I would prefer to use the OTEL operator. That is the purpose of the workshop.

Reply:
We should as well define the collector CR in a separate file with multiple versions (files e.g. `-N`). Based on the https://docs.google.com/document/d/11SJfXjt8_DU1IV7IWO-3GhMQ1kMtE5hNNbWJQEfcVOs/edit# the users will be changing the collector config

Reply:
Didn't we agree that we wanted to keep the backend opaque? The idea here is to imitate a backend that accepts all signals in otlp format. The data sources are also not named "tempo,loki,..." its just called "logs,metrics,traces".

Reply:
Means we have less struggle with configuring specific exporters and database endpoints (e.g. loki push stuff). If we want to forward logs, we simply forward otlp to `otel-collector.observability-backend.svc.cluster.local:4317`.

Reply:
Maybe we should rename the service and namespace to something like:
```
backend.observability.svc.cluster.local:4317
```
wdyt?

Reply:
I don't care much about the name, my comment was about why the collector deployment is not using the operator.

Reply:
y, that is intended. The idea is to mimic some kind of backend. Make it easy to send telemetry data there, without dealing with specific exporters for specific databases.

This pr aims for an arch like this:
![image](https://user-images.githubusercontent.com/10403402/229105586-cff51b43-4580-419a-aeee-2563bf74104d.png)

Reply:
But why? Then we would need to deal with all the different exporters.

Reply:
To have a single way of deploying the collector in the demo.  

Why are we deploying it by vanilla k8s manifest if we want to promote the operator?